concordium-consensus/src/Concordium/KonsensusV1/SkovMonad.hs

```diff
@@ -0,0 +1,568 @@
+    databaseHandlers = vcDisk
+
+instance (MonadTrans (SkovV1T pv)) where
+    lift = SkovV1T . lift
+
+instance Monad m => MonadState (SkovData pv) (SkovV1T pv m) where
+    state = SkovV1T . state . v1sSkovData
+    get = SkovV1T (use v1sSkovData)
+    put = SkovV1T . (v1sSkovData .=)
+
+deriving via
+    (PersistentBlockStateMonadHelper pv m)
+    instance
+        IsProtocolVersion pv => MonadProtocolVersion (SkovV1T pv m)
+
+deriving via
+    (PersistentBlockStateMonadHelper pv m)
+    instance
+        (IsProtocolVersion pv, MonadIO m) => AccountOperations (SkovV1T pv m)
+
+deriving via
+    (PersistentBlockStateMonadHelper pv m)
+    instance
+        (IsProtocolVersion pv, MonadIO m) => BlockStateQuery (SkovV1T pv m)
+
+deriving via
+    (PersistentBlockStateMonadHelper pv m)
+    instance
+        (IsProtocolVersion pv, MonadIO m) => BlockStateOperations (SkovV1T pv m)
+
+deriving via
+    (PersistentBlockStateMonadHelper pv m)
+    instance
+        (IsProtocolVersion pv, MonadIO m) => BlockStateStorage (SkovV1T pv m)
+
+deriving via
+    (DiskLLDBM pv (InnerSkovV1T pv m))
+    instance
+        ( IsProtocolVersion pv,
+          MonadIO m,
+          MonadCatch m,
+          MonadLogger m
+        ) =>
+        LowLevel.MonadTreeStateStore (SkovV1T pv m)
+
+instance Monad m => MonadBroadcast (SkovV1T pv m) where
+    sendTimeoutMessage tm = do
+        handler <- view sendTimeoutHandler
+        lift $ handler tm
+    sendQuorumMessage qm = do
+        handler <- view sendQuorumHandler
+        lift $ handler qm
+    sendBlock sb = do
+        handler <- view sendBlockHandler
+        lift $ handler sb
+
+instance Monad m => MonadConsensusEvent (SkovV1T pv m) where
+    onBlock bp = do
+        handler <- view onBlockHandler
+        lift $ handler bp
+    onFinalize fe bp = do
+        handler <- view onFinalizeHandler
+        lift $ handler fe bp
+    onPendingLive = do
+        handler <- view onPendingLiveHandler
+        lift handler
+
+instance (MonadIO m, MonadLogger m) => TimerMonad (SkovV1T pv m) where
+    type Timer (SkovV1T pv m) = ThreadTimer
+    onTimeout timeout a = do
+        ctx <- ask
+        liftIO $
+            makeThreadTimer timeout $ do
+                let handler (SomeException e) =
+                        _skovV1TUnliftIO ctx $
+                            logEvent Konsensus LLError $
+                                "Error in timer thread: " ++ show e
+                void (_skovV1TUnliftIO ctx a) `catchAll` handler
+    cancelTimer = liftIO . cancelThreadTimer
+
+instance
+    ( MonadIO m,
+      MonadLogger m,
+      MonadCatch m,
+      IsProtocolVersion pv,
+      IsConsensusV1 pv,
+      TimeMonad m
+    ) =>
+    MonadTimeout (SkovV1T pv m)
+    where
+    resetTimer dur = do
+        mTimer <- SkovV1T $ use v1sTimer
+        mapM_ cancelTimer mTimer
+        newTimer <- onTimeout (DelayFor $ durationToNominalDiffTime dur) uponTimeoutEvent
+        SkovV1T $ v1sTimer ?= newTimer
```

Comment:
Shouldn't this be `?=!`?

Or are things sufficiently evaluated?

Reply:
It should not be necessary, but I will put it anyway.